akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -110,6 +115,9 @@ private[remote] sealed trait HeaderBuilder {
   def setUid(u: Long): Unit
   def uid: Long
 
+  /** Metadata SPI, internally multiple metadata sections can be represented. */
+  def metadata: ByteString
```

Comment:
in any case we'll need a copy once, so ByteString we can get away by slicing for the sub parts